packages/select/src/common/listItemsProps.ts

```diff
@@ -124,6 +124,22 @@ export interface IListItemsProps<T> extends IProps {
      */
     onQueryChange?: (query: string, event?: React.ChangeEvent<HTMLInputElement>) => void;
 
+    /**
+     * If provided, allows new items to be created with the current query string.
+     * This is invoked when user interaction causes a new item to be created, either by pressing the `enter` key or
+     * by clicking on the "Create Item" option. It transforms a query string into an item type.
+     */
+    createItemFromQuery?: (query: string) => T;
```

Comment:
API mostly looks good but can we be a little more explicit and say `createNewItemFromQuery` / `createNewItemRenderer` ?

Reply:
Ack